src/common/bitfield.h

```diff
@@ -70,8 +73,9 @@ struct BitFieldContainer {
   };
 
  private:
-  common::Span<value_type> bits_;
-  static_assert(!std::is_signed<VT>::value, "Must use unsiged type as underlying storage.");
+  value_type* bits_{nullptr};
```

Comment:
Can you explain why we should use a raw pointer here? You mentioned the use of `std::vector<bool>` in `ColumnMatrix`. Does the span behave badly when `value_type == bool` ?

Reply:
It's used in hot loops, the check used in span can be expensive.